src/main/java/com/purbon/kafka/topology/api/ccloud/CCloudApi.java

```diff
@@ -0,0 +1,102 @@
+
+  private JulieHttpClient httpClient;
+
+  public CCloudApi(String baseServerUrl, Configuration config) {
+    this(new JulieHttpClient(baseServerUrl, Optional.of(config)), config);
+  }
+
+  public CCloudApi(JulieHttpClient httpClient, Configuration config) {
+    this.httpClient = httpClient;
+    var basicAuth = config.getConfluentCloudAuth();
+    this.httpClient.setBasicAuth(basicAuth);
+  }
+
+  public void createAcl(String clusterId, TopologyAclBinding binding) throws IOException {
+    String url = String.format("/kafka/v3/clusters/%s/acls", clusterId);
+    var request = new KafkaAclRequest(binding, String.format("%s%s", httpClient.baseUrl(), url));
+    httpClient.doPost(url, request.asJson());
+  }
+
+  public void deleteAcls(String clusterId, TopologyAclBinding binding) throws IOException {
+    String url = String.format("%s/kafka/v3/clusters/%s/acls", httpClient.baseUrl(), clusterId);
+    KafkaAclRequest request = new KafkaAclRequest(binding, url);
+    httpClient.doDelete(request.deleteUrl());
+  }
+
+  public ServiceAccount createServiceAccount(String sa) throws IOException {
+    return createServiceAccount(sa, "Managed by JulieOps");
+  }
+
+  public void deleteServiceAccount(String sa) throws IOException {
+    String url = String.format("https://api.confluent.cloud/iam/v2/service-accounts/%s", sa);
+    httpClient.doDelete(url);
+  }
+
+  public ServiceAccount createServiceAccount(String sa, String description) throws IOException {
+    String url = "https://api.confluent.cloud/iam/v2/service-accounts";
+    var request = new ServiceAccountRequest(sa, description);
+    String responseBody = httpClient.doPost(url, request.asJson());
+
+    ServiceAccountResponse response =
+        (ServiceAccountResponse) JSON.toObject(responseBody, ServiceAccountResponse.class);
+    return new ServiceAccount(
+        response.getId(),
+        response.getDisplay_name(),
+        response.getDescription(),
+        response.getMetadata().getResource_name());
+  }
+
+  public Set<ServiceAccount> listServiceAccounts() throws IOException {
+    String url = "https://api.confluent.cloud/iam/v2/service-accounts";
+    boolean finished;
+    Set<ServiceAccount> accounts = new HashSet<>();
+
+    do {
+      ListServiceAccountResponse response = getListServiceAccounts(url);
+      for (ServiceAccountResponse serviceAccountResponse : response.getData()) {
+        var resourceId = serviceAccountResponse.getMetadata().getResource_name();
+        var serviceAccount =
+            new ServiceAccount(
+                serviceAccountResponse.getId(),
+                serviceAccountResponse.getDisplay_name(),
+                serviceAccountResponse.getDescription(),
+                resourceId);
+        accounts.add(serviceAccount);
+      }
+
+      String nextUrl = response.getMetadata().getNext();
+      String lastUrl = response.getMetadata().getLast();
+      finished = nextUrl.equals(lastUrl);
```

Comment:
Would this stop condition mean that we would not pick up the lastUrl?

If I understand the Confluent Clound API docs correctly, we should stop when nextUrl == null.

Reply:
Amended as well, thanks a lot for your thoughts Sounie. Keep in mind this was a WiP pull requests anyhow ;-)